Lib/test/datetimetester.py

```diff
@@ -399,6 +400,11 @@ def test_harmless_mixed_comparison(self):
         self.assertIn(me, [1, 20, [], me])
         self.assertIn([], [me, 1, 20, []])
 
+        # Comparison over a type other that is not object's type should return
+        # NotImplemented and fallback to other.__eq__. In this case ANY.__eq__
+        # always returns True.
+        self.assertEqual(me, ANY)
```

Comment:
I also think it would be a good idea to check negative comparison (as is done earlier in this test), because the fallback path depends on which methods are implemented. See [this test in `dateutil`](https://github.com/dateutil/dateutil/blob/master/dateutil/test/test_tz.py#L714), for example.